src/native/corehost/hostmisc/pal.unix.cpp

```diff
@@ -341,7 +341,18 @@ bool get_extraction_base_parent_directory(pal::string_t& directory)
     // check for the POSIX standard environment variable
     if (pal::getenv(_X("HOME"), &directory))
     {
-        return is_read_write_able_directory(directory);
+        if (is_read_write_able_directory(directory))
+        {
+            return true;
+        }
+        else
+        {
+            trace::error(_X("Default extraction directory [%s] either doesn't exist or is not accessible for read/write."), directory.c_str());
+        }
+    }
+    else
+    {
+        trace::error(_X("Failed to determine default extraction location. Environment variable '$HOME' is not defined."));
```

Comment:
nit: for `$HOME`-less case, we can attempt `getpwuid_r` as done in libs:
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Native/Unix/System.Native/pal_networking.c#L3038-L3056
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/IO/PersistedFiles.Unix.cs#L125-L137
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Native/Unix/System.Native/pal_uid.c#L65-L68

Reply:
Thanks @am11. There's https://github.com/dotnet/runtime/issues/58464 which is basically tracking potential improvements in where to put the files. I think using other means to obtain the default location falls into that category as well. I wanted to keep this one very simple, solely as a diagnostics improvement (not changing actual functional behavior).

Reply:
Alright, makes sense. I made a small patch on top of your branch with this suggestion: https://github.com/am11/runtime/commit/0ae4966. Perhaps some variant of it could be considered. Will follow up once this PR is merged. :)